### ISSUE TITLE
Question about extracting cross-chain information using ESM-PPI

### ISSUE THREAD

**Issue by wyattxuanyang** [TencentAI4S/tfold#14]
Thank you for your great work!

As a user I have some questions. Suppose that there are four or five chains (eg. the TCR-pMHC complex, with the chains of A, B, M, N, P), could you please tell me whether I can still extract cross-chain information using ESM-PPI, since I notice TCR-pMHC complex is included in the PPI training process. And what are the exact meaning of the final inputs in the provided code here? (the output keys are `(['labl', 'mask', 'pred', 'sfea', 'pfea'])`)

Thank you! :)

```
import torch
import tfold

# Download the pre-trained model
model_path = tfold.model.esm_ppi_650m_ab()

# Load the model
model = tfold.model.PPIModel.restore(model_path)

# Prepare antibody sequences (can be single or multiple sequences)
data = [
        'QVQLVQSGAEVKKPGASVKVSCKASGYPFTSYGISWVRQAPGQGLEWMGWISTYNGNTNYAQKFQGRVTMTTDTSTTTGYMELRRLRSDDTAVYYCARDYTRGAWFGESLIGGFDNWGQGTLVTVSS', # Heavy chain
        'EIVLTQSPGTLSLSPGERATLSCRASQTVSSTSLAWYQQKPGQAPRLLIYGASSRATGIPDRFSGSGSGTDFTLTISRLEPEDFAVYYCQQHDTSLTFGGGTKVEIK' # Light chain
]
ppi_output = model(data)
```

**Comment by wufandi** [comment]
Thank you for your interest in our work! 

Theoretically, you can use ESM-PPI (TCR) to extract cross-chain information for the TCR-pMHC complex. However, I believe that the information provided by ESM-PPI for TCR-pMHC (A, B, M, N, P) is limited for a couple of reasons: 
1. There is a scarcity of paired TCR-pMHC data; 
2. 2. PLM-based language models may not effectively learn the interactions between TCR and pMHC.

Generally, I think that using ESM-PPI-TCR to extract cross-chain information is more reliable when dealing with unbound TCR systems (A, B) and unliganded pMHC systems (M, N, P).

**Comment by Wolkenwandler** [comment]
> Thank you for your great work!
> 
> As a user I have some questions. Suppose that there are four or five chains (eg. the TCR-pMHC complex, with the chains of A, B, M, N, P), could you please tell me whether I can still extract cross-chain information using ESM-PPI, since I notice TCR-pMHC complex is included in the PPI training process. And what are the exact meaning of the final inputs in the provided code here? (the output keys are `(['labl', 'mask', 'pred', 'sfea', 'pfea'])`)
> 
> Thank you! :)
> 
> ```
> import torch
> import tfold
> 
> # Download the pre-trained model
> model_path = tfold.model.esm_ppi_650m_ab()
> 
> # Load the model
> model = tfold.model.PPIModel.restore(model_path)
> 
> # Prepare antibody sequences (can be single or multiple sequences)
> data = [
>         'QVQLVQSGAEVKKPGASVKVSCKASGYPFTSYGISWVRQAPGQGLEWMGWISTYNGNTNYAQKFQGRVTMTTDTSTTTGYMELRRLRSDDTAVYYCARDYTRGAWFGESLIGGFDNWGQGTLVTVSS', # Heavy chain
>         'EIVLTQSPGTLSLSPGERATLSCRASQTVSSTSLAWYQQKPGQAPRLLIYGASSRATGIPDRFSGSGSGTDFTLTISRLEPEDFAVYYCQQHDTSLTFGGGTKVEIK' # Light chain
> ]
> ppi_output = model(data)
> ```

Each key in the outputs dictionary represents a different component of the model's output:  
* 'labl': Original token indices from tokn_mat_orig that represent the unmasked (original) amino acid sequences. These serve as ground truth labels for training.  
* 'mask': Binary mask tensor indicating which positions were masked during inference. It's used to identify which positions should be predicted in the masked language modeling task.  
* 'pred': The model's logits (raw prediction scores before softmax) for each position in the sequence. These are the actual predictions made by the model.  
* 'sfea': Single-residue features/embeddings extracted from the final layer representations. These are residue-level embeddings with dimension self.c_s for each amino acid position. 
* 'pfea': Pair features representing interactions between residues, derived from attention weights. These capture the relationships between each pair of residues in the sequence with dimension self.c_z.
